src/grisp_tools_deploy.erl

```diff
@@ -86,15 +86,22 @@ download(#{package := #{meta := Meta}} = State0) ->
     end,
     State2.
 
-extract(#{package := #{state := downloaded, file := File}} = State0) ->
+extract(#{package := #{state := State, file := File}} = State0) 
+        when (State == downloaded) or (State == not_modified) ->   
     #{paths := #{install := InstallPath}} = State0,
+    case filelib:is_dir(InstallPath) of
+        true -> event(State0, ['_skip']);
+        false -> extract_really(File, InstallPath, State0)
+    end;
+extract(State0) ->
+    event(State0, ['_skip']).
+
+extract_really(File, InstallPath, State0) ->
```

Comment:
Just a comment: usually these follow-up functions are prefixed with `do_` (like `do_extract`) but that's by no means an official thing.